draco/analysis/delay.py

```diff
@@ -1082,18 +1082,27 @@ def _evaluate(self, data_view, weight_view, out_cont, delays, channel_ind):
             data = data_view.local_array[lbi]
             weight = weight_view.local_array[lbi]
 
+            # Apply the window if requested
+            # Do this before applying data cut
+            # as that will change the data shape
+            if self.apply_window:
```

Comment:
Yeah this makes sense. I think the reason it got through is because with `freq_frac=-1` no frequency cuts are applied, so the data was unchanged along that axis.